src/compiler/diagnosticMessages.json

```diff
@@ -1476,7 +1476,14 @@
         "category": "Error",
         "code": 2505
     },
-
+    "A non-ambient class cannot be merged with an interface.": {
+        "category": "Error",
+        "code": 2506
+    },
+    "An interface cannot merge with a non-ambient class.": {
+        "category": "Error",
+        "code": 2507
+    },

```

Comment:
I would just pick one of these. Also, I would cancel out the 'non' and the 'cannot' to make an 'only'.